pstress/vault_test_setup.sh

```diff
@@ -0,0 +1,293 @@
+  echo 000a > serialfile
+  touch certindex
+  echo "[ ca ]" > ${WORKDIR}/certificates/vault-ca.conf
+  echo "default_ca = myca" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "[ myca ]" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "new_certs_dir = /tmp" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "unique_subject = no" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "certificate = ${WORKDIR}/certificates/root.cer" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "database = ${WORKDIR}/certificates/certindex" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "private_key = ${WORKDIR}/certificates/privkey.pem" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "serial = ${WORKDIR}/certificates/serialfile" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "default_days = 365" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "default_md = sha256" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "policy = myca_policy" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "x509_extensions = myca_extensions" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "copy_extensions = copy" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "[ myca_policy ]" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "commonName = supplied" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "stateOrProvinceName = supplied" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "countryName = supplied" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "emailAddress = optional" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "organizationName = supplied" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "organizationalUnitName = optional" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "[ myca_extensions ]" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "basicConstraints = CA:false" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "subjectKeyIdentifier = hash" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "authorityKeyIdentifier = keyid:always" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "subjectAltName = IP:${VAULT_IP}" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "keyUsage = digitalSignature,keyEncipherment" >> ${WORKDIR}/certificates/vault-ca.conf
+  echo "extendedKeyUsage = serverAuth" >> ${WORKDIR}/certificates/vault-ca.conf
+  openssl ca -batch -config vault-ca.conf -notext -in vault.csr -out vault.crt >/dev/null 2>&1
+  popd >/dev/null
+  if [[ ${GEN_CERTS_ONLY} -eq 1 ]]; then
+    echo "Generated SSL certificates and exiting..."
+    exit 0
+  fi
+fi
+
+if [[ ${DL_VAULT} -eq 1 ]]; then
+  if [[ -x "${WORKDIR}/vault" ]]; then
+    echo -e "\n===== Vault binary already present, skipping download ====="
+  else
+    echo -e "\n===== Downloading vault binary ====="
+    VAULT_URL=$(${SCRIPT_PWD}/get_download_link.sh --product=vault)
```

Comment:
A suggestion. is there a possibility to get the getdownload.sh sh script from Github? ex: wget "https://github.com/Percona-QA/percona-qa/blob/master/get_download_link.sh"

One of the reasons im thinking is that the script has get some links been fixed, and if that would be better fetched dynamically, what do you think?

Reply:
I think it is better to retain a copy of get_download_link.sh in our pstress repository. In future, we do not know if we are going to maintain Percona-QA repo. If the script is part of pstress repo, the ownership of maintaining this file will be with pstress developers.